rest-service/manager_rest/resource_manager.py

```diff
@@ -1381,6 +1383,105 @@ def _finalize_create_deployment(self, deployment: models.Deployment):
         # RD-1603 will move this:
         self.create_deployment_schedules(deployment, deployment.blueprint.plan)
 
+    @staticmethod
+    def get_deployment_parents_from_labels(labels):
+        parents = []
+        for label, value in labels:
+            if label == 'csys-obj-parent':
+                parents.append(value)
+        return parents
+
+    def verify_deployment_parent_labels(self, parents, deployment_id):
+        if not parents:
+            return
+        missing_parents = []
+        result = self.sm.list(
+            models.Deployment,
+            include=['id'],
+            filters={'id': lambda col: col.in_(parents)}
+        ).items
+        _existing_parents = [_parent[0] for _parent in result]
+        for parent in parents:
+            if parent not in _existing_parents:
+                missing_parents.append(parent)
+
+        if missing_parents:
+            raise manager_exceptions.DeploymentParentNotFound(
+                'Deployment {0}: is referencing deployments'
+                ' using label `csys-obj-parent` that are not existed, '
```

Comment:
that are not existed -> that do not exist

Reply:
done